core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java

```diff
@@ -72,6 +112,72 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String fieldInQuery = this.fieldName;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            fieldInQuery = fieldType.names().indexName();
+        }
+        SpanQuery innerQuery = (SpanQuery) ((QueryBuilder) this.queryBuilder).toQuery(parseContext);
```

Comment:
Thinking more about it, it is probably more natural to make your proposed change and have the `QueryBuilder` member. Then also update the constructor to:

```
public <T extends QueryBuilder & SpanQueryBuilder> FieldMaskingSpanQueryBuilder(T queryBuilder, String field) {
```

so we make sure that the argument is both a query builder and a span query builder. Thoughts?